pkg/metadata/process_test.go

```diff
@@ -0,0 +1,201 @@
+			cgroups: []procfs.Cgroup{{
+				HierarchyID: 0,
+				Controllers: []string{},
+				Path:        "/system.slice/systemd-journald.service",
+			}},
+			wantIndex: 0,
+		},
+		{
+			name: "deeper cgroup path",
+			cgroups: []procfs.Cgroup{{
+				HierarchyID: 0,
+				Controllers: []string{},
+				Path:        "/user.slice/user-1000.slice/user@1000.service/init.scope",
+			}},
+			wantIndex: 0,
+		},
+		{
+			name: "root cgroup path",
+			cgroups: []procfs.Cgroup{{
+				HierarchyID: 0,
+				Controllers: []string{},
+				Path:        "/",
+			}},
+			wantIndex: 0,
+		},
+		{
+			name: "extract cpu controller from multiple cgroup controllers",
+			cgroups: []procfs.Cgroup{
+				{
+					HierarchyID: 11,
+					Controllers: []string{"hugetlb"},
+					Path:        "/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod1ff39434b35faeef64159d11e3f96024.slice/docker-a.scope",
+				},
+				{
+					HierarchyID: 10,
+					Controllers: []string{"pids"},
+					Path:        "/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod1ff39434b35faeef64159d11e3f96024.slice/docker-a.scope",
+				},
+				{
+					HierarchyID: 9,
+					Controllers: []string{"blkio"},
+					Path:        "/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod1ff39434b35faeef64159d11e3f96024.slice/docker-a.scope",
+				},
+				{
+					HierarchyID: 8,
+					Controllers: []string{"net_cls", "net_prio"},
+					Path:        "/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod1ff39434b35faeef64159d11e3f96024.slice/docker-a.scope",
+				},
+				{
+					HierarchyID: 7,
+					Controllers: []string{"cpuset"},
+					Path:        "/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod1ff39434b35faeef64159d11e3f96024.slice/docker-a.scope",
+				},
+				{
+					HierarchyID: 6,
+					Controllers: []string{"devices"},
+					Path:        "/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod1ff39434b35faeef64159d11e3f96024.slice/docker-a.scope",
+				},
+				{
+					HierarchyID: 5,
+					Controllers: []string{"memory"},
+					Path:        "/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod1ff39434b35faeef64159d11e3f96024.slice/docker-a.scope",
+				},
+				{
+					HierarchyID: 4,
+					Controllers: []string{"perf_event"},
+					Path:        "/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod1ff39434b35faeef64159d11e3f96024.slice/docker-a.scope",
+				},
+				{
+					HierarchyID: 3,
+					Controllers: []string{"cpu", "cpuacct"},
+					Path:        "/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod1ff39434b35faeef64159d11e3f96024.slice/docker-09af509f3db677a2275723fc71bff3d9b6d19e4d404c44822f2262f700adcd4b.scope",
+				},
+				{
+					HierarchyID: 2,
+					Controllers: []string{"freezer"},
+					Path:        "/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod1ff39434b35faeef64159d11e3f96024.slice/docker-a.scope",
+				},
+				{
+					HierarchyID: 1,
+					Controllers: []string{"name=systemd"},
```

Comment:
Is this case real? It is not listed as a supported controller:
https://systemd.io/CGROUP_DELEGATION/#controller-support

Reply:
cc @javierhonduco I assume you've added this example.

Reply:
Sorry, somehow missed this! I think this was added by Maxime, but might be wrong 😄 . Either way, I am not sure about how real this case is 😿